### ISSUE TITLE
Radio button set to readonly (invalid HTML5)

### ISSUE THREAD

**Issue by KevinBongart** [heartcombo/simple_form#1730]
## Current behavior

I have this wrapper in my app, [coming from this repo](https://github.com/heartcombo/simple_form-bootstrap/blob/a8063cbdedda32025753ac3cb0b182f7853a1497/config/initializers/simple_form_bootstrap.rb#L78-L88):

```rb
config.wrappers :vertical_collection, item_wrapper_class: 'form-check', tag: 'fieldset', class: 'form-group', error_class: 'form-group-invalid', valid_class: 'form-group-valid' do |b|
  b.use :html5
  b.optional :readonly
  b.wrapper :legend_tag, tag: 'legend', class: 'col-form-label pt-0' do |ba|
    ba.use :label_text
  end
  b.use :input, class: 'form-check-input', error_class: 'is-invalid', valid_class: 'is-valid'
  b.use :full_error, wrap_with: { tag: 'div', class: 'invalid-feedback d-block' }
  b.use :hint, wrap_with: { tag: 'small', class: 'form-text text-muted' }
end

…

config.wrapper_mappings = {
  …
  radio_buttons: :vertical_collection,
  …
}
```

I'm not entirely sure how `simple_form` uses the `readonly` option, but I recently ran into an issue ([on Stack Overflow](https://stackoverflow.com/questions/66624562/capybara-is-not-choosing-radio-button-without-javascript/66630198#66630198)) where Capybara couldn't interact with one of the two radio buttons without a JavaScript driver (such as `selenium_chrome_headless`). It looks that is because the resulting HTML has a `readonly="readonly"` on one of the two radio buttons, [which apparently is not valid HTML](https://stackoverflow.com/questions/1953017/why-cant-radio-buttons-be-readonly).

Here's the ERB code:

```erb
<%= simple_form_for(@foo) do |f| %>
  <%= f.simple_fields_for :bar do |b| %>
    <%= p.input :baz, as: :radio_buttons, label: '' %>
  <% end %>

  <%= f.button :submit %>
<% end %>
```

The `baz` field is a boolean attribute on the `bar` class, and the line generates this HTML (note the `readonly` on the second input):

```html
<fieldset class="form-group radio_buttons optional foo_bar_baz form-group-valid">
  <legend class="col-form-label pt-0"></legend>
  <input type="hidden" name="foo[bar_attributes][baz]" value="">

  <div class="form-check">
    <input class="form-check-input is-valid radio_buttons optional" type="radio" value="true" checked="checked" name="foo[bar_attributes][baz]" id="foo_bar_attributes_baz_true">
    <label class="collection_radio_buttons" for="foo_bar_attributes_baz_true">Yes</label>
  </div>

  <div class="form-check">
    <input class="form-check-input is-valid radio_buttons optional" readonly="readonly" type="radio" value="false" name="foo[bar_attributes][baz]" id="foo_bar_attributes_baz_false">
    <label class="collection_radio_buttons" for="foo_bar_attributes_baz_false">No</label>
  </div>
</fieldset>
```

Capybara has since then been fixed to ignore the `readonly` attributes on radio buttons, but I'm still wondering about where in the stack the issue originates, _if that's an issue at all_, and what I can do to work around it.

## Environment

- Ruby **3.0.0p0**
- Rails **6.1.3**
- Simple Form **5.1.0**

## Expected behavior

I believe there shouldn't be a `readonly="readonly"` attribute on the radio button unless explicitly passed in, and even then, since it's apparently not valid HTML, maybe some validation should prevent it. If you believe otherwise, I'd love to hear the rational behind it and figure out a workaround.

**Comment by nashby** [comment]
Hey @KevinBongart. You can find more info about this issue here https://github.com/heartcombo/simple_form/issues/1257. It's not really optimal but that's how it works at the moment. 

TL;DR Just remove `b.optional :readonly` from your configuration and use `readonly` component when you need
